packages/router/src/bindings/contractReader/index.ts

```diff
@@ -100,6 +101,47 @@ export const handleActiveTransactions = async (
   _requestContext?: RequestContext,
 ) => {
   const { logger } = getContext();
+
+  const tmpLiquidityMap: Map<string, BigNumber> = new Map();
+
+  // calculate pending balances
+  for (const transaction of transactions) {
+    // if transfer is receiver prepared, just increase pending liquidity
+    if (transaction.status === CrosschainTransactionStatus.ReceiverPrepared) {
+      const liquidityKey = transaction.crosschainTx.invariant.receivingChainId
+        .toString()
+        .concat("-")
+        .concat(transaction.crosschainTx.invariant.receivingAssetId);
+      const prevLiquidity = tmpLiquidityMap.get(liquidityKey);
+      const pendingAmount = BigNumber.from(transaction.crosschainTx.receiving?.amount);
+
+      if (tmpLiquidityMap.has(liquidityKey) && prevLiquidity) {
+        tmpLiquidityMap.set(liquidityKey, prevLiquidity.add(pendingAmount));
+      } else {
+        tmpLiquidityMap.set(
+          transaction.crosschainTx.invariant.receivingChainId
+            .toString()
+            .concat("-")
+            .concat(transaction.crosschainTx.invariant.receivingAssetId),
+          BigNumber.from(pendingAmount),
+        );
+      }
+    } else if (transaction.status === CrosschainTransactionStatus.ReceiverFulfilled) {
+      tmpLiquidityMap.set(
+        transaction.crosschainTx.invariant.sendingChainId
+          .toString()
+          .concat("-")
+          .concat(transaction.crosschainTx.invariant.sendingAssetId),
+        BigNumber.from(transaction.crosschainTx.sending?.amount),
+      );
```

Comment:
this will overwrite everything that exists for that asset, and only save it for that transaction when it should be cumulative